install_config/router/customized_haproxy_router.adoc

```diff
@@ -13,48 +13,665 @@ toc::[]
+Note that if there is only one variable, it is assigned the element;
+this is opposite to the convention in Go range clauses.
+
+A variable's scope extends to the *end* action of the control structure
+(*if*, *with*, or *range*) in which it is declared, or to the end of the
+template if there is no such control structure. A template invocation does
+not inherit variables from the point of its invocation.
+
+
+[[go-template-functions]]
+=== Go Template Functions
+
+.Predefined Global Functions
+[cols="2*", options="header"]
+|===
+|Function Name | Meaning
+|`*and*`| Returns the boolean AND of its arguments by returning the
+	first empty argument or the last argument, that is,
+	"and x y" behaves as "if x then y else x". All the
+	arguments are evaluated.
+
+|`*index*`| Returns the result of indexing its first argument by the
+	following arguments. Thus "index x 1 2 3" is, in Go syntax,
+	x[1][2][3]. Each indexed item must be a map, slice, or array.
+
+|`*len*`| Returns the integer length of its argument.
+|`*not*`| Returns the boolean negation of its single argument.
+|`*or*`| Returns the boolean OR of its arguments by returning the
+	first non-empty argument or the last argument, that is,
+	"or x y" behaves as "if x then x else y". All the
+	arguments are evaluated.
+|===
+
+.Comparison Functions
+[cols="2*", options="header"]
+|===
+|Function | Meaning
+|`*eq*` arg1 arg2 | Returns the boolean truth of arg1 == arg2
+|`*ne*` arg1 arg2 | Returns the boolean truth of arg1 != arg2
+|`*lt*` arg1 arg2 | Returns the boolean truth of arg1 < arg2
+|`*le*` arg1 arg2 | Returns the boolean truth of arg1 <= arg2
+|`*gt*` arg1 arg2 | Returns the boolean truth of arg1 > arg2
+|`*ge*` arg1 arg2 | Returns the boolean truth of arg1 >= arg2
+|===
+The boolean functions take any zero value to be false and a non-zero value
+to be true.
+
+.Template Router Functions
+[cols="2*", options="header"]
+|===
+|Type | Meaning
+|`*endpointsForAlias*` | returns the list of valid endpoints
+|`*env*` | tries to get the named environment variable from the pod if it
+can't it returns the second argument.
+|`*matchPattern*` | the firt argument is a string that contains the regular
+expression, the second argument is the variable to test. If the regular
+expression matches the variable, return true otherwise reeturn false.
+|`*isInteger*` | determines if a given variable is an integer
+|`*matchValues*` | compares a given string to a list of allowed strings
+|`*genSubdomainWildcardRegexp*` | generates a regular expression matching the
+subdomain for hosts (and paths) with a wildcard policy
+|`*generateRouteRegexp*` | generates a regular expression matching the route
+hosts (and paths). The first argument is the host name, the second is the path
+and the third is a wildcard boolean.
+|`*genCertificateHostName*` | generates host name to use for serving/matching
+certificates.  First argument is the host name and the second is the wildcard
+bolean.
+|===
```

Comment:
No I meant like a golang function signature for all the helper functions ala: 
`func env(name, defaultValue string) string` 
`func matchPattern(pattern, s string) bool`

etc. and add some info re: the function (or just point to the examples you have below) ... might be a good resource for someone who understands golang.